manifests/zone.pp

```diff
@@ -16,13 +16,39 @@
 # @param manage_file_name
 #   Whether to set the file parameter in the zone file.
 #
-# @param update_policy_rules
+# @param update_policy
 #   This can be used to specifiy additional update policy rules in the
 #   following format
 #   { '<KEY_NAME' => {'matchtype' => '<VALUE>', 'tname' => '<VALUE>', 'rr' => 'VALUE' } }
 #   Example {'foreman_key' => {'matchtype' => 'zonesub', 'rr' => 'ANY'}}
 #   tname and rr are optional
 #
+# @param target_views Array[String]
+# @param zonetype String
+# @param soa String
+# @param reverse Boolean
+# @param ttl String
+# @param refresh Integer
+# @param update_retry Integer
+# @param expire Integer
+# @param negttl Integer
+# @param serial Integer
+# @param masters Array
+# @param allow_transfer Array
+# @param allow_query Array
+# @param also_notify Array
+# @param zone String
+# @param contact Optional[String]
+# @param zonefilepath Stdlib::Absolutepath
+# @param filename String
+# @param forward Enum['first', 'only']
+# @param forwarders Array
+# @param dns_notify Optional[Enum['yes', 'no', 'explicit']]
+# @param key_directory Optional[Stdlib::Absolutepath]
+# @param inline_signing Optional[Enum['yes', 'no']]
+# @param dnssec_secure_to_insecure Optional[Enum['yes', 'no']]
+# @param auto_dnssec Optional[Enum['allow', 'maintain', 'off']]
```

Comment:
I think append the type this way doesn't really work. puppet-strings infers the type if there is one. That means you can leave it off. Ideally we would actually have documentation what they do.

Reply:
You are right, the type doesn't need to be there.

I haven't added what the params do because I don't really know. My intention was only to fix the mistake with `update_policy_rules`. Adding the params is only an extra here. You can look at it as a "baby step" :-)